pycbc/inference/io/dynesty.py

```diff
@@ -25,12 +25,96 @@
 """
 from pycbc.io.hdf import (dump_state, load_state)
 from .base_nested_sampler import BaseNestedSamplerFile
+from .posterior import write_samples_to_file, read_raw_samples_from_file
 
-class DynestyFile(BaseNestedSamplerFile):
+import numpy
+import argparse
+
+class CommonNestedMetadataIO(object):
+    """Provides functions for reading/writing dynesty metadata to file.
+    """
+    @staticmethod
+    def extra_args_parser(parser=None, skip_args=None, **kwargs):
+        """Create a parser to parse sampler-specific arguments for loading
+        samples.
+        Parameters
+        ----------
+        parser : argparse.ArgumentParser, optional
+            Instead of creating a parser, add arguments to the given one. If
+            none provided, will create one.
+        skip_args : list, optional
+            Don't parse the given options. Options should be given as the
+            option string, minus the '--'. For example,
+            ``skip_args=['iteration']`` would cause the ``--iteration``
+            argument not to be included.
+        \**kwargs :
+            All other keyword arguments are passed to the parser that is
+            created.
+        Returns
+        -------
+        parser : argparse.ArgumentParser
+            An argument parser with th extra arguments added.
+        actions : list of argparse.Action
+            A list of the actions that were added.
+        """
+        if parser is None:
+            parser = argparse.ArgumentParser(**kwargs)
+        elif kwargs:
+            raise ValueError("No other keyword arguments should be provded if "
+                             "a parser is provided.")
+        if skip_args is None:
+            skip_args = []
+        actions = []
+
+        if 'raw_samples' not in skip_args:
+            act = parser.add_argument(
+                "--raw-samples", type=numpy.array, default=False,
+                help="Raw samples are the unweighted samples obtained from "
+                     "dynesty sampler."
```

Comment:
"dynesty sampler" -> "nested sampler"

Reply:
But this particular function deals specifically with dynesty sampler.

Reply:
Ok, fair enough